features/netsocket/RAWSocket.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2019 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "RAWSocket.h"
+#include "Timer.h"
+#include "mbed_assert.h"
+
+RAWSocket::RAWSocket()
```

Comment:
This is a big duplication of UDPSocket (another one after `CellularNonIPSocket`). In this case it's so similar, because it's NSAPI, that I'm thinking there should be an `InternetDatagramSocket` base class to subsume the commonality, that both this and `UDPSocket` inherit from.

What do others think?

Reply:
Yes InternetDatagramSocket can be a base class, if Application needs to create an ICMP socket they can use this class.